docs/application/native/guides/compliations/watchface-complication.md

```diff
@@ -0,0 +1,341 @@
+}
+```
+ 
+2. Watchface can make the complication_candidates_list_h to identify the editable features to request to editor.
+Watchface can request to edit the features by using watchface_editable_request_edit().
+
+3. The watchface editor can draw highlight to give the information what is editing.
+To inform the position of editable features, watchface_editable_highlight_h should be used.
+The highlight handle can be created by watchface_editable_hilight_create() function with watchface shape type.
+
+There are following shape types.
+
+| Shape types | Description |
+|-------------|-------------|
+| WATCHFACE_EDITABLE_SHAPE_TYPE_CIRCLE | circle |
+| WATCHFACE_EDITABLE_SHAPE_TYPE_RECT | rectangle |
+
+
+The position of editable feature can be set by using watchface_editable_highlight_set_geometry().
+The parameters of this function mean the position(x,y) and size(width, height).
+
+
+
+> **Note**
+>
+> watchface_editable_highlight_h must be released after used by using watchface_editable_highlight_destroy().
+
+
+4. According to the editable feature, watchface_editable_add_design_element() and watchface_editable_add_complication()
+ are used to add editable features to the watchface_editable_container_h.
+
+
+```cpp
+static void _watchface_editable_edit_ready_cb(watchface_editable_container_h handle,
+		const char *editor_appid, void *user_data)
+{
+	bundle *candidate1;
+	bundle *candidate2;
+	complication_h comp;
+	watchface_editable_highlight_h hi;
+	complication_candidates_list_h candidates_list = NULL;
+ 
+	watchface_editable_candidates_list_create(&candidates_list);
+ 
+	candidate1 = bundle_create();
+	bundle_add_str(candidate1, "TEST_COLOR", "RED");
+	watchface_editable_candidates_list_add(candidates_list, candidate1);
+	bundle_free(candidate1);
+ 
+	candidate2 = bundle_create();
+	bundle_add_str(candidate2, "TEST_COLOR", "YELLOW");
+	watchface_editable_candidates_list_add(candidates_list, candidate2);
+	bundle_free(candidate2);
+ 
+	watchface_editable_highlight_create(&hi, WATCHFACE_EDITABLE_SHAPE_TYPE_CIRCLE);
+	watchface_editable_highlight_set_geometry(hi, 50, 50, 100, 100);
+	watchface_editable_add_design_element(handle, COLOR_EDIT, cur_idx, candidates_list, hi, "Color");
+	watchface_editable_highlight_destroy(hi);
+ 
+	watchface_editable_candidates_list_destroy(list_handle);
+ 
+	watchface_editable_highlight_create(&hi, WATCHFACE_EDITABLE_SHAPE_TYPE_CIRCLE);
+        watchface_editable_highlight_set_geometry(hi, 100, 100, 100, 100);
+	watchface_editable_add_complication(handle, comp_id, comp, hi);
+	watchface_editable_highlight_destroy(hi);
+ 
+	watchface_editable_request_edit(handle, _watchface_editable_update_requested_cb, user_data);
+}
+```
+
+5. If some features are changed from editor, watchface_editable_update_requested_cb() is called.
+Watchface can check the watchface_editable_edit_state_e from the parameter of that callback to get the edit state.
+

```

Comment:
If some features are changed from editor, `watchface_editable_update_requested_cb()` is called.
Watchface can verify `watchface_editable_edit_state_e` from the parameter of that callback to get the edit state.